From 6c03dbdcbf876b41ad5a82e2248fadb056525bef Mon Sep 17 00:00:00 2001
From: Roger Leigh <rleigh@debian.org>
Date: Mon, 22 Nov 2010 21:03:24 +0000
Subject: [PATCH 2/6] dh: Use $(filter) rather than $(findstring)

$(findstring) can match partial strings and so is unreliable when a
package builds several binary packages and one package contains the
name of another package within its name.  In these cases,
$(findstring) can return a partial match which leads to problems
(performing unwanted actions which could lead to build failure, for
example).

$(filter) matches the entire string in the wordlist, so is a
reliable replacement for $(findstring).

Signed-off-by: Roger Leigh <rleigh@debian.org>
---
 dh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/dh b/dh
index 3e014fb..0b07973 100755
--- a/dh
+++ b/dh
@@ -235,7 +235,7 @@ L<dh_listpackages(1)> to test what is being built. For example:
 	
 	override_dh_fixperms:
 		dh_fixperms
-	ifneq (,$(findstring foo, $(shell dh_listpackages)))
+	ifneq (,$(filter foo, $(shell dh_listpackages)))
 		chmod 4755 debian/foo/usr/bin/foo
 	endif
 
-- 
1.7.2.3

